store/tikv/gcworker/gc_worker.go

```diff
@@ -522,33 +626,100 @@ func doGC(ctx goctx.Context, store tikv.Storage, safePoint uint64, identifier st
 	// Sleep to wait for all other tidb instances update their safepoint cache.
 	time.Sleep(gcSafePointCacheInterval)
 
-	log.Infof("[gc worker] %s start gc, safePoint: %v.", identifier, safePoint)
+	log.Infof("[gc worker] %s start gc, concurrency %v, safePoint: %v.", identifier, concurrency, safePoint)
 	startTime := time.Now()
 	successRegions := 0
 	failedRegions := 0
 
 	ticker := time.NewTicker(gcJobLogTickInterval)
 	defer ticker.Stop()
 
-	bo := tikv.NewBackoffer(tikv.GcOneRegionMaxBackoff, ctx)
+	// create task queue and task result queue, and start task workers
+	gcTaskCh := make(chan *gcTask, concurrency)
+	gcResultCh := make(chan *gcResult, concurrency*2)
+	wg := sync.WaitGroup{}
+	for i := 0; i < concurrency; i++ {
+		w := newGCTaskWorker(store, gcTaskCh, gcResultCh, &wg)
+		wg.Add(1)
+		go w.run()
+	}
+
 	var key []byte
+	defer func() {
+		close(gcTaskCh)
+		count := 0
+		for taskRes := range gcResultCh {
+			if taskRes == nil {
+				count++
+				if count == concurrency {
+					close(gcResultCh)
```

Comment:
We don't need to deal with `gcResultCh`, its capacity is large enough, no task worker will be blocked on sending to this channel.